mlrun/model_monitoring/model_endpoint.py

```diff
@@ -0,0 +1,132 @@
+        self.algorithm = algorithm
+        self.monitor_configuration = monitor_configuration or {}
+        self.active = active
+        self.monitoring_mode = monitoring_mode
+
+
+class ModelEndpointStatus(mlrun.model.ModelObj):
+    def __init__(
+        self,
+        feature_stats: Optional[dict] = None,
+        current_stats: Optional[dict] = None,
+        first_request: Optional[str] = "",
+        last_request: Optional[str] = "",
+        error_count: Optional[int] = 0,
+        drift_status: Optional[str] = "",
+        drift_measures: Optional[dict] = None,
+        metrics: Optional[Dict[str, Dict[str, Any]]] = None,
+        features: Optional[List[Dict[str, Any]]] = None,
+        children: Optional[List[str]] = None,
+        children_uids: Optional[List[str]] = None,
+        endpoint_type: Optional[EndpointType] = EndpointType.NODE_EP,
+        monitoring_feature_set_uri: Optional[str] = "",
+    ):
+        self.feature_stats = feature_stats or {}
+        self.current_stats = current_stats or {}
+        self.first_request = first_request
+        self.last_request = last_request
+        self.error_count = error_count
+        self.drift_status = drift_status
+        self.drift_measures = drift_measures or {}
+        self.features = features or []
+        self.children = children or []
+        self.children_uids = children_uids or []
+        self.endpoint_type = endpoint_type or EndpointType.NODE_EP
+        self.monitoring_feature_set_uri = monitoring_feature_set_uri
+        if metrics is None:
+            self.metrics = {
+                EventKeyMetrics.GENERIC: {
+                    EventLiveStats.LATENCY_AVG_1H: 0,
+                    EventLiveStats.PREDICTIONS_PER_SECOND: 0,
+                }
+            }
+
+
+class ModelEndpoint(mlrun.model.ModelObj):
+    kind = "model-endpoint"
+    _dict_fields = ["kind", "metadata", "spec", "status"]
+
+    def __init__(self, uid: str = None):
+        self._status: ModelEndpointStatus = ModelEndpointStatus()
+        self._spec: ModelEndpointSpec = ModelEndpointSpec()
+        self._metadata: mlrun.model.VersionedObjMetadata = (
+            mlrun.model.VersionedObjMetadata()
+        )
+        self.uid = uid
+
+    @property
+    def status(self) -> ModelEndpointStatus:
+        return self._status
+
+    @status.setter
+    def status(self, status):
+        self._status = self._verify_dict(status, "status", ModelEndpointStatus)
+
+    @property
+    def spec(self) -> ModelEndpointSpec:
+        return self._spec
+
+    @spec.setter
+    def spec(self, spec):
+        self._spec = self._verify_dict(spec, "spec", ModelEndpointSpec)
+
+    @property
+    def metadata(self) -> mlrun.model.VersionedObjMetadata:
+        if not self.uid:
+            self.uid = create_model_endpoint_id(
+                function_uri=self.spec.function_uri, versioned_model=self.spec.model
+            )
+            self.uid = str(self.uid)
+        self._metadata.uid = self.uid
```

Comment:
Right. Actually because this is the `ModelObj` which used by the client side we don't need this here at all. The creator should generate the uid by itself (as done in `routers.py` and `v2_serving.py`).